app/controllers/CompanyController.scala

```diff
@@ -1,65 +1,66 @@
-
-  val logger: Logger = Logger(this.getClass)
-
-  def getCompanies(search: String, postalCode: Option[String], maxCount: Int) = UnsecuredAction.async { implicit request =>
-    var request = ws
-      .url(s"https://entreprise.data.gouv.fr/api/sirene/v1/full_text/$search")
-      .addHttpHeaders("Accept" -> "application/json", "Content-Type" -> "application/json")
-
-    if (postalCode.isDefined) {
-      request = request.addQueryStringParameters("code_postal" -> postalCode.get)
-    }
-    
-    request = request.addQueryStringParameters("per_page" -> maxCount.toString)
 
-    request.get().flatMap(
-      response => {
-        response.status match {
-          case NOT_FOUND => Future(NotFound(response.json))
-          case status if isServerError(status) =>
-            logger.error(s"getCompanies [$search, $postalCode, $maxCount] - $response")
-            Future(InternalServerError(response.json))
-          case _ => Future(Ok(response.json))
-        }
+@Singleton
+class CompanyController @Inject()(
+                                val userRepository: UserRepository,
+                                val companyRepository: CompanyRepository,
+                                val accessTokenRepository: AccessTokenRepository,
+                                val silhouette: Silhouette[AuthEnv]
+                              )(implicit ec: ExecutionContext)
+ extends BaseCompanyController {
+  def findCompany(q: String) = SecuredAction(WithRole(UserRoles.Admin)).async { implicit request =>
+    for {
+      company <- q match {
+        case q if q.matches("[a-zA-Z0-9]{8}-[a-zA-Z0-9]{4}") => companyRepository.findByShortId(q)
+        case q if q.matches("[0-9]{14}") => companyRepository.findBySiret(SIRET(q))
+        case q => companyRepository.findByName(q)
       }
-    );
-
+    } yield company.map(c => Ok(Json.obj(
+      "id" -> c.id,
+      "siret" -> c.siret,
+      "name"  -> c.name,
+      "address"  -> c.address,
+      "postalCode"  -> c.postalCode
+    ))).getOrElse(NotFound)
   }
 
-  def getCompaniesBySiret(siret: String, maxCount: Int) = UnsecuredAction.async { implicit request =>
-
-    logger.debug(s"getCompaniesBySiret [$siret]")
-
-    var request = ws
-      .url(s"https://entreprise.data.gouv.fr/api/sirene/v1/siret/$siret")
-      .addHttpHeaders("Accept" -> "application/json", "Content-Type" -> "application/json")
-
-    request = request.addQueryStringParameters("per_page" -> maxCount.toString)
-
-    request.get().flatMap(
-      response => response.status match {
-        case NOT_FOUND => Future(NotFound(response.json))
-        case status if isServerError(status) =>
-          logger.error(s"getCompaniesBySiret [$siret] - $response")
-          Future(InternalServerError(response.json))
-        case _ => Future(Ok(response.json))
-      }
-    );
-
+  def companyDetails(siret: String) = SecuredAction(WithRole(UserRoles.Admin)).async { implicit request =>
```

Comment:
Pourquoi pas par `uuid`(évite en plus de gérer les `getOrElse` dans le `for each`) ?

Reply:
J'ai fait comme ça par souci de cohérence avec les autres API qui utilisent le SIRET, mais si un jour on a des entreprises sans SIRET...

Cela dit, mais avec le UUID pourquoi n'aurait-on pas de `getOrElse` ?

Reply:
Oui t'as raison rien à voir avec le `getOrElse`